src/hooks.server.ts

```diff
@@ -107,22 +104,33 @@ const handleGenericActionRequest: Handle = async ({ event, resolve }) => {
 			return new Response(FORBIDDEN_ACCESS_RESPONSE, { status: 403 });
 		}
 		// Request header configuration.
-		const userTenantHeader = event.request.headers.get(USER_TENANT_HEADER);
+		const userTenantHeaderHash = event.cookies.get(USER_TENANT_HEADER) ?? '';
+
 		const user = session.user as CustomUserSession;
-		const currentUserData = user.tenantUsers.find(
-			(tenantUser) => tenantUser.id === userTenantHeader
+		const tenantUserValidations = await Promise.all(
+			user.tenantUsers.map(async (tenantUser) => {
```

Comment:
Yo creo que eras tú el que estaba excéptico con las allamadas asincrónicas dentro de un map. Ojo que yo no le veo nada de malo excepto q le matas el paralelismo antes de tiempo.

Reply:
El problema es que las operaciones de alto consumo no deberian hacerse en dentro de un for...of o for convencional, si no esperas el valor en orden pudieras usar sin peligro cualquier operacion async dentro de un map, ak no necesito el orden solo que el map me devuelva el arreglo de promesas para darle al Promise.all. Despues discutamos esto pero creo que mi comentario es valido.

Reply:
El paralelismo ak fue para aprovechar el async behaviour de la funcion bcrypt.compare